### PR TITLE
mytarget: raise exception if http status code != 200

### PR DESCRIPTION
**Pull request by benoitbazouin**
### Description

- The goal of this PR is to raise an exception in the reader when we have an api response with http status != 200. 
- When this exception is raised, the code breaks without retrying and there is no file returned in the end. I did this because in our case, we got only '404' errors, with the error message "resources not found"
- The error message is logged, and the url requested too

### CODE REVIEW

**Review by bibimorlet — APPROVED** [review]

File: ack/readers/mytarget/reader.py [hunk 1]
```diff
@@ -168,7 +169,11 @@ def split_content_by_date(self, content: List[Dict[str, Any]]):
                 content_by_date.append(new_line)
         yield from content_by_date
 
-    @retry(wait=wait_exponential(multiplier=1, min=1, max=600), stop=stop_after_delay(600))
+    @retry(
+        wait=wait_exponential(multiplier=1, min=1, max=600),
```
Comment (tom-grivaud) [inline ack/readers/mytarget/reader.py:173]: Looking at the time it took for each id, can you confirm that this 10 mins limit still makes sense ?

**Review by tom-grivaud — APPROVED** [review]
LGTM 👍

File: ack/readers/mytarget/reader.py [hunk 2]
```diff
@@ -182,10 +187,15 @@ def __get_response(self, name_content: str, offset=0) -> Dict[str, Any]:
         """
         parameters = self.__generate_params_dict(name_content, offset=offset)
         request = self.__create_request(name_content, parameters)
-        resp = requests.get(**request).json()
-        if "items" not in resp.keys():
-            raise MissingItemsInResponse("Can't retrieve any item from this response")
-        return resp
+        resp = requests.get(**request)
+        if resp.status_code != HTTP_STATUS_OK:
```
Comment (tom-grivaud) [inline ack/readers/mytarget/reader.py:191]: Are you sure we never get another 2XX code here ? I'm wondering if getting a 202 for example could occur and then raise an exception while it's not really an error. Maybe need to check the API doc.

File: ack/readers/mytarget/reader.py (hunk 1 above)
Comment (benoitbazouin) [reply ack/readers/mytarget/reader.py:173]: Yes, it still makes sense

File: ack/readers/mytarget/reader.py (hunk 2 above)
Comment (benoitbazouin) [reply ack/readers/mytarget/reader.py:191]: The document of the API does not explain wether or not we can get another 2XX code. For our use case, we only got 200.

**Pull request merged**